cve_bin_tool/nvd_api.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Retrieval access and of NVD entries using NVD Automatic CVE Retrieval
+"""
+
+import asyncio
+import gzip
+import json
+import math
+import os
+from datetime import datetime
+from logging import Logger
+from typing import Union
+
+import aiohttp
+from rich.progress import Progress, track
+
+from cve_bin_tool.async_utils import RateLimiter
+from cve_bin_tool.error_handler import ErrorMode
+from cve_bin_tool.log import LOGGER
+
+NVD_FILENAME_TEMPLATE = "nvdcve-1.1-{}.json.gz"
+DISK_LOCATION_DEFAULT = os.path.join(os.path.expanduser("~"), ".cache", "cve-bin-tool")
+
+
+FEED = "https://services.nvd.nist.gov/rest/json/cves/1.0"
+
+PAGESIZE = 2000
+MAX_FAIL = 5
+# Interval in seconds between successive requests
+INTERVAL_PERIOD = 2
+
+
+def filter_by_id(cve, update_data):
+    """"Function to filter out duplicate CVE entries in case of incremental update"""
```

Comment:
```suggestion
    """Function to filter out duplicate CVE entries in case of incremental update"""
```
There are 5 doc strings in this file that start with 4 double quotes instead of 3.